src/core/model/reducer.ts

```diff
@@ -92,6 +135,46 @@ export const combinedReducer = (state: EditorState, action: CombinedAction) => {
   return state;
 };
 
+interface RemoveElementError {
+  id: 'removeElementError';
+  root: any;
+  element: any;
+}
+
+const isRemoveElementError = <T>(
+  result: T | RemoveElementError
+): result is RemoveElementError =>
+  (result as RemoveElementError).id === 'removeElementError';
+
+const removeUiElement = (
+  elementToMove: LinkedUISchemaElement,
+  uiSchema: LinkedUISchemaElement
+): LinkedUISchemaElement | RemoveElementError => {
+  const errorCase: RemoveElementError = {
+    id: 'removeElementError',
+    root: uiSchema,
+    element: elementToMove,
+  };
+  const oldParentPath = calculatePath(
+    getRoot(elementToMove.parent as LinkedUISchemaElement),
+    elementToMove.parent as LinkedUISchemaElement
+  );
+  const index = (elementToMove.parent as Layout).elements.indexOf(
```

Comment:
I would move the index just before the splice as the value is not needed before that